### PR TITLE
Feature/smooth animations

### PR DESCRIPTION
**Pull request by dan-rpm**
Changes:

- `<App />` component split into work envs (just to avoid huge components)
-  `<MultipleAnimationModel />` component added (it supports multiple animations as props like `Record<string, string>` and `activeAnimation` - which is a key of animations object)  (video below)

https://github.com/user-attachments/assets/91b6cea0-fd90-43e6-a2a8-7b8d5a03b0e1

-  `useLoader()` added to `<AnimationModel />` component to avoid multiple loading of model and/or animations (before/after below)

https://github.com/user-attachments/assets/0d38cea9-1872-4be2-9e4f-d27dab0e9590

https://github.com/user-attachments/assets/282f5ecd-cab1-4c05-b2b0-5bed6c6c51e9

### CODE REVIEW

File: src/components/Models/AnimationModel/AnimationModel.component.tsx [hunk 1]
```diff
@@ -38,7 +39,7 @@ export const AnimationModel: FC<AnimationModelProps> = ({
     setAnimationRunning(false);
   };
 
-  const { scene } = useGltfLoader(modelSrc);
+  const { scene } = useLoader(GLTFLoader, String(modelSrc));
```
Comment (Jarvv) [inline src/components/Models/AnimationModel/AnimationModel.component.tsx:42]: We need the loader to be compatible with mesh compression and DRACO.

Take a look at our current loader on what to add:
https://github.com/readyplayerme/visage/blob/bd2fc81d29fecbfce2a8e17a1086fe46520174c2/src/services/Models.service.tsx#L214-L219

Comment (dan-rpm) [reply src/components/Models/AnimationModel/AnimationModel.component.tsx:42]: nice, didnt see it

File: src/components/Models/MultipleAnimationModel/MultipleAnimationModel.component.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+import { BaseModelProps } from 'src/types';
+
+export interface MultipleAnimationModelProps extends BaseModelProps {
+  modelSrc: string | Blob;
+  animations: Record<string, string>;
+  activeAnimation: string;
+  scale?: number;
+}
+
+export const MultipleAnimationModel: FC<MultipleAnimationModelProps> = ({
```
Comment (Jarvv) [inline src/components/Models/MultipleAnimationModel/MultipleAnimationModel.component.tsx:16]: Should we also include the other props from AnimationModel, such as emotion?

File: src/components/Models/MultipleAnimationModel/MultipleAnimationModel.component.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    const prevAction = activeAction;
+
+    if (prevAction) {
+      newAction.reset();
+      newAction.crossFadeFrom(prevAction, 0.5, true);
+    }
+
+    newAction.play();
+    setActiveAction(newAction);
+  }, [activeAnimation, loadedAnimations, activeAction]);
```
Comment (Jarvv) [inline src/components/Models/MultipleAnimationModel/MultipleAnimationModel.component.tsx:72]: If are using emotion, [and nodes](https://github.com/readyplayerme/visage/blob/main/src/components/Models/AnimationModel/AnimationModel.component.tsx#L42), then we can add nodes.Armature to this dependancy.

```suggestion
  }, [activeAnimation, loadedAnimations, activeAction, nodes.Armature]);
```

At the moment, if you change the model url then the animation won't apply to the new loaded model.

Comment (dan-rpm) [reply src/components/Models/MultipleAnimationModel/MultipleAnimationModel.component.tsx:72]: yep, will add emotions also

Comment (dan-rpm) [reply src/components/Models/MultipleAnimationModel/MultipleAnimationModel.component.tsx:72]: image
Fixed

**Review by Jarvv — APPROVED** [review]

**Pull request merged**